### PR TITLE
Integrate UnlockProvider with app

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR moves handling of user account log in to `providerMiddleware`, which instantiates `UnlockProvider` with the user's decrypted wallet, then sets the account address in state and dispatches `providerReady`.

This PR does _not_ add `UnlockProvider` to the config, since I think we don't want it to be used yet (once we add `UnlockProvider` to config, it will always be present and there will not be an error message for people without MetaMask; we don't want that to happen until `UnlockProvider` is more capable)

To do a live test of `UnlockProvider` in this branch, or after it is merged, you have to import `UnlockProvider` in the config, disable the `Web3` and `HTTP` providers, and after the `readOnlyProvider` is set, do `providers['Unlock'] = new UnlockProvider({ readOnlyProvider })`.

Then you can go to the signup or login page and try it out.

# Issues

Refs #2414
# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/middlewares/providerMiddleware.ts [hunk 1]
```diff
@@ -22,13 +33,41 @@ function initializeProvider(provider: { enable?: () => any }, dispatch: any) {
...
+  const { key, emailAddress, password } = action
+  try {
+    await unlockProvider.connect({ key, password })
+
+    const address = unlockProvider.wallet.address
+    dispatch(setAccount({ address }))
+    dispatch(setEncryptedPrivateKey(key, emailAddress))
+    dispatch(providerReady())
+  } catch (_) {
+    // TODO: password isn't the only thing that can go wrong here...
```
Comment (cellog) [inline unlock-app/src/middlewares/providerMiddleware.ts]: can we console.error the error? it makes debugging ever so much more possible

Comment (cnasc) [reply unlock-app/src/middlewares/providerMiddleware.ts]: Do our lint rules not complain about that? There are actually a bunch of places where we could log that out if it's allowed.

Comment (cellog) [reply unlock-app/src/middlewares/providerMiddleware.ts]: oh if you do:

```js
// eslint-disable-next-line no-console
console.error(e)
```

Then lint doesn't complain :)

Comment (cellog) [reply unlock-app/src/middlewares/providerMiddleware.ts]: I think this is a good use case for breaking the lint rules, in other words

**Pull request merged**